ragna/core/_rag.py

```diff
@@ -261,22 +276,12 @@ def _unpack_chat_params(
             for fn, model in component_models.items()
         }
 
-    async def _run(self, fn: Callable[..., Union[T, Awaitable[T]]], *args: Any) -> T:
-        kwargs = self._unpacked_params[fn]
-        if inspect.iscoroutinefunction(fn):
-            fn = cast(Callable[..., Awaitable[T]], fn)
-            return await fn(*args, **kwargs)
-        else:
-            fn = cast(Callable[..., T], fn)
-            return await anyio.to_thread.run_sync(
-                functools.partial(fn, *args, **kwargs)
-            )
-
-    async def __aenter__(self) -> Chat:
-        await self.prepare()
-        return self
+    # TODO: Do we need this?
```

Comment:
See explanation above why I want to keep the `prepare` and in turn this as well.